src/containers/AssetForm.js

```diff
@@ -422,6 +405,32 @@ class AssetForm extends Component {
   }
 }
 
-const mapDispatchToProps = { handleMessage: snackbarOpen };
+AssetForm.propTypes = {
+  match: PropTypes.object.isRequired,
+  history: PropTypes.object.isRequired,
+  snackbarOpen: PropTypes.func.isRequired,
+  getAsset: PropTypes.func.isRequired,
+  createAsset: PropTypes.func.isRequired,
+  updateAsset: PropTypes.func.isRequired,
+  assetUrl: PropTypes.string,
+  asset: PropTypes.object,
+};
+
+const mapDispatchToProps = { snackbarOpen, getAsset, createAsset, updateAsset };
+
+const mapStateToProps = ({ assets } , props) => {
+
+  let assetUrl = assets.url; // for resolving the url post-save
```

Comment:
I don't understand how this is used. AFACIT, if we are creating an asset and have come from half way down the asset list ``assetUrl`` will be something like ``https://example.com/assets/cursor=sfnfjkndsfjkndjfsk&next=fdsndsjnfjdsk&ordering=-updated_at&department=UIS``. Then we lookup this URL in assetsByUrl which (correctly) finds nothing. Then, on mount, the component has ``this.props.asset === undefined`` and so calls (line 113) ``this.props.getAsset(this.props.assetUrl)`` which attempts to fetch an asset from the random asset list URL?